### PR TITLE
Task74 create role base PO list rendering in dashboard

### PR DESCRIPTION
**Pull request by mllrdev-jerry-bayoneta**

### CODE REVIEW

File: src/core/enums/user-roles.enum.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+enum UserRole {
+  APPROVER = 'APPROVER',
+  REQUESTOR = 'REQUESTOR',
+  ADMIN = 'ADMIN',
+  SUPER_ADMIN = 'SUPER_ADMIN'
+}
+
+export default UserRole;
```
Comment (mllrdev-carlo-medel) [inline src/core/enums/user-roles.enum.ts]: Rename file name to user-role.enum.ts

File: src/core/interfaces/user.interface.ts [hunk 2]
```diff
@@ -1,9 +1,15 @@
+import UserRole from "@/core/enums/user-roles.enum";
+
 export interface IUser {
-    firstName: string
-    middleName: string
-    lastName: string
-    email: string
-    role: string
+
```
Comment (mllrdev-carlo-medel) [inline src/core/interfaces/user.interface.ts:2]:
```suggestion
```

File: src/core/services/jwt.service.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import IUser from "@/core/interfaces/user.interface";
+
+const token = sessionStorage.getItem('token') as string
+function getTokenProperties() {
+  const payload: IUser = JSON.parse(JSON.stringify(jwtDecode(token)))
+  console.log(payload);
+  return payload
+}
+
+export default getTokenProperties
```
Comment (mllrdev-carlo-medel) [inline src/core/services/jwt.service.ts]:
```suggestion
const userPayload: String = sessionStorage.getItem('user')
function getUserFromPayload() {
  const user: IUser = JSON.parse(JSON.stringify(jwtDecode(userPayload)))
  return user
}

export default getTokenProperties
```

File: src/core/services/user.service.ts [hunk 4]
```diff
@@ -17,7 +17,8 @@ const login = async (loginDetails: ILoginRequest) => {
             const loginResponse: ILoginResponse = response.data;
             console.log(loginResponse.isSuccessful)
             if (response.data.isSuccessful) {
-                sessionStorage.setItem('token', 'Bearer ' + response.data.token)
+                sessionStorage.setItem('key', 'Bearer ' + response.data.token)
+                sessionStorage.setItem('token', response.data.token)
```
Comment (mllrdev-carlo-medel) [inline src/core/services/user.service.ts]:
```suggestion
                sessionStorage.setItem('token', `Bearerr${esponse.data.token}`)
                sessionStorage.setItem('user', response.data.token)
```

File: src/core/services/client.service.ts [hunk 5]
```diff
@@ -2,14 +2,14 @@ import environment from "@/environments/environment";
 import axios from "axios";
 import IClientRequest from "../interfaces/client-request.interface";
 
-const token = sessionStorage.getItem('token')
+const key = sessionStorage.getItem('key')
```
Comment (mllrdev-carlo-medel) [inline src/core/services/client.service.ts]:
```suggestion
const token = sessionStorage.getItem('token')
```

File: src/pages/Home.vue [hunk 6]
```diff
@@ -70,30 +70,36 @@ export default defineComponent({
...
+          orders.value = value || [];
+        });
+      }
+      
+      if( role.value === UserRole.APPROVER ) {
+        orderService.approver.getAll().then((value: IPurchaseOrderResponse['simple'][]) => {
+          orders.value = value || [];          
+        }); 
+      }
+
```
Comment (mllrdev-carlo-medel) [inline src/pages/Home.vue]: use switch case for this.

File: src/core/interfaces/user.interface.ts (hunk 2 above)
Comment (mllrdev-jerry-bayoneta) [reply src/core/interfaces/user.interface.ts:2]: done

File: src/core/services/client.service.ts (hunk 5 above)
Comment (mllrdev-jerry-bayoneta) [reply src/core/services/client.service.ts]: done

File: src/core/services/jwt.service.ts (hunk 3 above)
Comment (mllrdev-jerry-bayoneta) [reply src/core/services/jwt.service.ts]: done

File: src/core/services/user.service.ts (hunk 4 above)
Comment (mllrdev-jerry-bayoneta) [reply src/core/services/user.service.ts]: done

File: src/pages/Home.vue (hunk 6 above)
Comment (mllrdev-jerry-bayoneta) [reply src/pages/Home.vue]: done

**Review by mllrdev-carlo-medel — APPROVED** [review]

**Pull request merged**